File: CSE_4116/Lab_1/transpositional_cipher.cpp
#include<bits/stdc++.h>
using namespace std;
int main() {
    int row, col;
    cout << "Enter Row Number:";
    cin >> row;
    cout << "Enter Col Number:";
    cin >> col;
    char plainMatrix[row][col];
    getchar();
    cout << "Enter Your Text According to your format\n";
    for (int i = 0;i < row;i++) {
        string one_row;
        getline(cin, one_row);
        for (int j = 0;j < col;j++)plainMatrix[i][j] = one_row[j];
    }
    char cipherMatrix[col][row];
    for (int j = 0;j < col;j++) {
        for (int i = 0;i < row;i++) {
            cipherMatrix[j][i] = plainMatrix[i][j];
        }
    }
    cout << "Plain Matrix\n\n";
    for (int i = 0;i < row;i++) {
        for (int j = 0;j < col;j++) {
            cout << plainMatrix[i][j];
        }
        cout << endl;
    }
    cout << "\n\nCipher Matrix\n\n";
    for (int i = 0;i < col;i++) {
        for (int j = 0;j < row;j++) {
            cout << cipherMatrix[i][j];
        }
        cout << endl;
    }

}

File: CSE_4116/Lab_1/transpositional_cipher.py
row = int(input("Enter Row Number:"))
col = int(input("Enter Column Number:"))
print("Enter Your Text in Matrix Format")
actualData=[]
cipherData=[]
data = []
for i in range(row):
    text = input()
    if len(text)!=col:
        print("Please Maintain Correct Data Format.")
        exit()
    actualData.append(text)
for j in range(col):
    new_row=[]
    for i in range(row):
        s = actualData[i][j]
        if(s.islower()):
            s = s.upper()
        elif(s.isupper()):
            s = s.lower()
        new_row.append(s)
    cipherData.append(''.join(new_row))
print("\nPlain Text in Matrix Form:\n")
for x in actualData:
    print(x)
print("")
print("Cipher Text after Transposition:\n")
for x in cipherData:
    print(x)
